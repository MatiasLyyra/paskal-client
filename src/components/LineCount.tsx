import React from 'react';
import './LineCount.scss';

type IProps = {
  lines: number,
};
const LineCount: React.FC<IProps> = props => (
  <ul className="LineCount-lines">
    {Array.from(Array(props.lines).keys()).map(i => (
      <li key={i}>{i + 1}.</li>
    ))}
  </ul>
);
export default LineCount;
