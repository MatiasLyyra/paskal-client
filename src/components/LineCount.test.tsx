import React from 'react';
import LineCount from './LineCount';
import {shallow} from 'enzyme';

it('renders renders 0 lines correctly', () => {
  const wrapper = shallow(<LineCount lines={0}></LineCount>);
  expect(wrapper).toMatchSnapshot();
});
it('renders renders n lines correctly', () => {
  const wrapper = shallow(<LineCount lines={12}></LineCount>);
  expect(wrapper).toMatchSnapshot();
});
