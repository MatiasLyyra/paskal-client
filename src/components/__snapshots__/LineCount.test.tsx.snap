// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders renders 0 lines correctly 1`] = `
<ul
  className="LineCount-lines"
/>
`;

exports[`renders renders n lines correctly 1`] = `
<ul
  className="LineCount-lines"
>
  <li
    key="0"
  >
    1
    .
  </li>
  <li
    key="1"
  >
    2
    .
  </li>
  <li
    key="2"
  >
    3
    .
  </li>
  <li
    key="3"
  >
    4
    .
  </li>
  <li
    key="4"
  >
    5
    .
  </li>
  <li
    key="5"
  >
    6
    .
  </li>
  <li
    key="6"
  >
    7
    .
  </li>
  <li
    key="7"
  >
    8
    .
  </li>
  <li
    key="8"
  >
    9
    .
  </li>
  <li
    key="9"
  >
    10
    .
  </li>
  <li
    key="10"
  >
    11
    .
  </li>
  <li
    key="11"
  >
    12
    .
  </li>
</ul>
`;
